steps/base.py

```diff
@@ -24,6 +38,95 @@ def __init__(self,
                  load_saved_output=False,
                  save_graph=False,
                  force_fitting=False):
+        """
+        Args:
+            name (int): Step name. Each step in a pipeline needs to have a unique name.
+                Transformers, and Step outputs will be persisted/cached/saved under this exact name.
+            transformer (obj): Step instance or object that inherits from BaseTransformer.
```

Comment:
@grzes314 it is either BaseTransformer or Step how would you handle that?

Reply:
Do we really have a usecase when we would like to put a Step here instead of Transformer? Anyway, if we incorporate Michal's recent proposition, Step and Transformer will be unified, so let's forget it.